packages/devextreme-angular/src/http/ajax.ts

```diff
@@ -0,0 +1,278 @@
+/* eslint-disable @typescript-eslint/no-floating-promises */
+import { Deferred, DeferredObj } from 'devextreme/core/utils/deferred';
+import {
+  HttpClient, HttpEventType, HttpParams, HttpEvent, HttpErrorResponse, HttpResponse,
+} from '@angular/common/http';
+import { throwError, Subject } from 'rxjs';
+import { takeUntil, timeoutWith } from 'rxjs/operators';
+import { isDefined } from 'devextreme/core/utils/type';
+import { getWindow } from 'devextreme/core/utils/window';
+import {
+  isCrossDomain,
+  evalCrossDomainScript,
+  getRequestOptions,
+  getJsonpCallbackName as getCallbackNameAndPatchOptions,
+  getRequestHeaders as getAjaxRequestHeaders,
+  getAcceptHeader,
+  evalScript,
+} from 'devextreme/core/utils/ajax_utils';
+
+type Result = Promise<any> & { abort: () => void };
+type DeferredResult = DeferredObj<any>;
+interface Options {
+  url: string;
+  [key: string]: any;
+}
+
+interface XHRSurrogate {
+  type?: string;
+  aborted: boolean;
+  abort: () => void;
+}
+
+const PARSER_ERROR = 'parsererror';
+const SUCCESS = 'success';
+const ERROR = 'error';
+const NO_CONTENT = 'nocontent';
+const TIMEOUT = 'timeout';
+const STATUS_ABORT = 0;
+const CONTENT_TYPE = 'Content-Type';
+const URLENCODED = 'application/x-www-form-urlencoded';
+
+function assignResponseProps(xhrSurrogate: XHRSurrogate, response: HttpResponse<any> | HttpErrorResponse) {
+  const getResponseHeader = (name: string) => response.headers.get(name);
+
+  function makeResponseText() {
+    const body = 'error' in response ? response.error : response.body;
+
+    if (typeof body !== 'string' || String(getResponseHeader(CONTENT_TYPE)).startsWith('application/json')) {
+      return JSON.stringify(body);
+    }
+
+    return body;
+  }
+
+  Object.assign(xhrSurrogate, {
+    status: response.status,
+    statusText: response.statusText,
+    getResponseHeader,
+    responseText: makeResponseText(),
+  });
+
+  return xhrSurrogate;
+}
+
+function isGetMethod(options: Options) {
+  return (options.method || 'get').toLowerCase() === 'get';
+}
+
+function isCacheNeed(options: Options) {
+  if (options.cache === undefined) {
```

Comment:
Seems that we can refactor it as
```
if (options.cache) { 
    return options.cache;
}

return !(isUsedScript(options) || isUsedJSONP(options));
```

Reply:
options.cache can be false, if so we no need calculate it